src/build/app/mod.rs

```diff
@@ -93,15 +89,88 @@ pub struct App<'b> {
     pub(crate) template: Option<&'b str>,
     pub(crate) settings: AppFlags,
     pub(crate) g_settings: AppFlags,
-    #[doc(hidden)]
-    pub args: MKeyMap<'b>,
-    #[doc(hidden)]
-    pub subcommands: Vec<App<'b>>,
+    pub(crate) args: MKeyMap<'b>,
+    pub(crate) subcommands: Vec<App<'b>>,
     pub(crate) replacers: HashMap<&'b str, &'b [&'b str]>,
     pub(crate) groups: Vec<ArgGroup<'b>>,
     pub(crate) help_headings: Vec<Option<&'b str>>,
 }
 
+impl<'b> App<'b> {
+    /// Get the name of the app
+    pub fn get_name(&self) -> &str {
+        &self.name
+    }
+
+    /// Get the name of the binary
+    pub fn get_bin_name(&self) -> Option<&str> {
+        self.bin_name.as_deref()
+    }
+
+    /// Set binary name. Uses `&mut self` instead of `self`
+    pub fn set_bin_name<S: Into<String>>(&mut self, name: S) {
+        self.bin_name = Some(name.into());
+    }
+
+    /// Get the help message specified via [`App::about`]
+    pub fn get_about(&self) -> Option<&str> {
+        self.about.as_deref()
+    }
+
+    /// Iterate through the *visible* aliases for this subcommand.
+    pub fn get_visible_aliases(&self) -> impl Iterator<Item = &str> {
+        self.aliases
+            .as_ref()
+            .into_iter()
+            .flat_map(|aliases| aliases.iter().filter(|(_, vis)| *vis).map(|a| a.0))
+    }
+
+    /// Iterate through the set of *all* the aliases for this subcommand, both visible and hidden.
+    pub fn get_all_aliases(&self) -> impl Iterator<Item = &str> {
+        self.aliases
+            .as_ref()
+            .into_iter()
+            .flat_map(|aliases| aliases.iter().map(|a| a.0))
+    }
+
+    /// Get the list of subcommands
+    pub fn get_subcommands(&self) -> &[App<'b>] {
+        &self.subcommands
+    }
+
+    /// Get the list of subcommands
+    pub fn get_subcommands_mut(&mut self) -> &mut [App<'b>] {
+        &mut self.subcommands
+    }
+
+    /// Get the list of arguments
+    pub fn get_arguments(&self) -> &[Arg<'b>] {
+        &self.args.args
+    }
+
+    /// Get the list of arguments the given argument conflicts with
+    ///
+    /// ### Panics
+    ///
+    /// Panics if the given arg conflicts with an argument that is unknown to this application
+    pub fn get_arg_conflicts_with<'a, 'x, 'y>(&'a self, arg: &'x Arg<'y>) -> Vec<&Arg<'b>> // FIXME: This could probably have been an iterator
```

Comment:
I don't know. I was receiving really cryptic errors without them, so I just assigned all the lifetimes explicitly. Let's just merge it because it works, we can rethink this later